### PR TITLE
Fix benchmarks and add more cases.

### PR DESCRIPTION
**Pull request by mathetake**
Some of the benchmarks have been broken as we didn't 
check the errors. This commit fixes it, add more cases
and produces the actual benchmark results.

___

```
goarch: amd64
cpu: AMD Ryzen 9 3950X 16-Core Processor            
BenchmarkEngines
BenchmarkEngines/naivevm
BenchmarkEngines/naivevm/base64_5_per_exec
BenchmarkEngines/naivevm/base64_5_per_exec-32           1000000000               0.0007209 ns/op               0 B/op          0 allocs/op
BenchmarkEngines/naivevm/base64_100_per_exec
BenchmarkEngines/naivevm/base64_100_per_exec-32         1000000000               0.01021 ns/op         0 B/op          0 allocs/op
BenchmarkEngines/naivevm/base64_10000_per_exec
BenchmarkEngines/naivevm/base64_10000_per_exec-32       1000000000               0.5875 ns/op          0 B/op          0 allocs/op
BenchmarkEngines/naivevm/fibofor_5
BenchmarkEngines/naivevm/fibofor_5-32                      88412             14147 ns/op            9224 B/op        380 allocs/op
BenchmarkEngines/naivevm/fibofor_10
BenchmarkEngines/naivevm/fibofor_10-32                      6082            167432 ns/op          108936 B/op       4511 allocs/op
BenchmarkEngines/naivevm/fibofor_20
BenchmarkEngines/naivevm/fibofor_20-32                        56          20171810 ns/op        13473914 B/op     558218 allocs/op
BenchmarkEngines/naivevm/string_manipulation_size_50
BenchmarkEngines/naivevm/string_manipulation_size_50-32                      214           5451473 ns/op         3536602 B/op     146660 allocs/op
BenchmarkEngines/naivevm/string_manipulation_size_100
BenchmarkEngines/naivevm/string_manipulation_size_100-32                      79          15338381 ns/op         9842499 B/op     406379 allocs/op
BenchmarkEngines/naivevm/string_manipulation_size_1000
BenchmarkEngines/naivevm/string_manipulation_size_1000-32                      2         924341970 ns/op        602264464 B/op  24639795 allocs/op
BenchmarkEngines/naivevm/reverse_array_size_500
BenchmarkEngines/naivevm/reverse_array_size_500-32                          1034           1136004 ns/op          755544 B/op      30876 allocs/op
BenchmarkEngines/naivevm/reverse_array_size_1000
BenchmarkEngines/naivevm/reverse_array_size_1000-32                          501           2158362 ns/op         1508296 B/op      61626 allocs/op
BenchmarkEngines/naivevm/reverse_array_size_10000
BenchmarkEngines/naivevm/reverse_array_size_10000-32                          48          21937936 ns/op        15057797 B/op     615126 allocs/op
BenchmarkEngines/naivevm/random_mat_mul_size_5
BenchmarkEngines/naivevm/random_mat_mul_size_5-32                            746           1614151 ns/op         1059720 B/op      45503 allocs/op
BenchmarkEngines/naivevm/random_mat_mul_size_10
BenchmarkEngines/naivevm/random_mat_mul_size_10-32                           157           7612253 ns/op         5061602 B/op     215029 allocs/op
BenchmarkEngines/naivevm/random_mat_mul_size_100
BenchmarkEngines/naivevm/random_mat_mul_size_100-32                            1        4734519289 ns/op        3246427296 B/op 133368916 allocs/op
BenchmarkEngines/wazeroir
BenchmarkEngines/wazeroir/base64_5_per_exec
BenchmarkEngines/wazeroir/base64_5_per_exec-32                          1000000000               0.0000669 ns/op               0 B/op          0 allocs/op
BenchmarkEngines/wazeroir/base64_100_per_exec
BenchmarkEngines/wazeroir/base64_100_per_exec-32                        1000000000               0.0007913 ns/op               0 B/op          0 allocs/op
BenchmarkEngines/wazeroir/base64_10000_per_exec
BenchmarkEngines/wazeroir/base64_10000_per_exec-32                      1000000000               0.05185 ns/op         0 B/op          0 allocs/op
BenchmarkEngines/wazeroir/fibofor_5
BenchmarkEngines/wazeroir/fibofor_5-32                                    853510              1298 ns/op             144 B/op         10 allocs/op
BenchmarkEngines/wazeroir/fibofor_10
BenchmarkEngines/wazeroir/fibofor_10-32                                    84042             13943 ns/op            1440 B/op         91 allocs/op
BenchmarkEngines/wazeroir/fibofor_20
BenchmarkEngines/wazeroir/fibofor_20-32                                      710           1714837 ns/op          175153 B/op      10948 allocs/op
BenchmarkEngines/wazeroir/string_manipulation_size_50
BenchmarkEngines/wazeroir/string_manipulation_size_50-32                    2751            367486 ns/op           34444 B/op        911 allocs/op
BenchmarkEngines/wazeroir/string_manipulation_size_100
BenchmarkEngines/wazeroir/string_manipulation_size_100-32                   1292            923810 ns/op           83649 B/op       1811 allocs/op
BenchmarkEngines/wazeroir/string_manipulation_size_1000
BenchmarkEngines/wazeroir/string_manipulation_size_1000-32                    25          49036675 ns/op         5807610 B/op      18011 allocs/op
BenchmarkEngines/wazeroir/reverse_array_size_500
BenchmarkEngines/wazeroir/reverse_array_size_500-32                        21375             58553 ns/op           12648 B/op          3 allocs/op
BenchmarkEngines/wazeroir/reverse_array_size_1000
BenchmarkEngines/wazeroir/reverse_array_size_1000-32                       10590            114361 ns/op              40 B/op          3 allocs/op
BenchmarkEngines/wazeroir/reverse_array_size_10000
BenchmarkEngines/wazeroir/reverse_array_size_10000-32                       1020           1227607 ns/op          645119 B/op          3 allocs/op
BenchmarkEngines/wazeroir/random_mat_mul_size_5
BenchmarkEngines/wazeroir/random_mat_mul_size_5-32                         10677            110478 ns/op            2920 B/op        183 allocs/op
BenchmarkEngines/wazeroir/random_mat_mul_size_10
BenchmarkEngines/wazeroir/random_mat_mul_size_10-32                         2278            491795 ns/op            8776 B/op        549 allocs/op
BenchmarkEngines/wazeroir/random_mat_mul_size_100
BenchmarkEngines/wazeroir/random_mat_mul_size_100-32                           4         281132028 ns/op          659944 B/op      41247 allocs/op
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -7,11 +7,11 @@ bench:
 
 .PHONY: build.bench
 build.bench:
-	tinygo build -o bench/case/case.wasm -scheduler=none -target=wasi bench/case/case.go
+	tinygo build -o bench/case/case.wasm -gc=leaking -scheduler=none -target=wasi bench/case/case.go
```
Comment (mathetake) [inline Makefile:10]: This is a temporary workaround on the bug of TinyGo's conservative GC (enabled by default). I will raise an issue to TinyGo repo later.

**Comment by mathetake** [comment]
huh this time filesystem has started failing even with wasmtime..

**Comment by codefromthecrypt** [comment]
🕵️

File: examples/wasm/file_system.go [hunk 2]
```diff
@@ -1,28 +0,0 @@
-package main
```
Comment (mathetake) [inline examples/wasm/file_system.go:1]: This doesn't work even with Wasmtime. So removing for now to not only prioritize the benchmarks, but also we will not focus on WASI until we finish JIT.

**Comment by mathetake** [comment]
merging as this is a bit urgent!

File: examples/wasm/file_system.go (hunk 2 above)
Comment (codefromthecrypt) [reply examples/wasm/file_system.go:1]: yep pick battles!

**Pull request merged**